wandb/apis/public.py

```diff
@@ -4235,9 +4235,30 @@ def id(self):
     def file_count(self):
         return self._attrs["fileCount"]
 
+    @property
+    def source_version(self):
+        """
+        Returns:
+            (str) The artifact's version index under its parent artifact collection. This will return
+            a string with the format "v{number}".
+        """
+        return f"v{self._sequence_version_index}"
+
     @property
     def version(self):
-        return "v%d" % self._version_index
+        """
```

Comment:
While this is technically correct - it does depend on the convenience that version index is encoded as a `v#` alias. I think a more stable implementation would be to modify the GQL to leverage the `versionIndex` edge which accepts an optional collection name. This would allow you to directly fetch the logical version index without relying on the alias data structure.

Reply:
Looks like the `versionIndex` resolver is incorrect when taking in the `artifactCollectionName`. It assumes that this collection lives in the same project as the artifact version, so this will require a backend schema change as well.